angularjs-viewhead.js

```diff
@@ -1,76 +1,66 @@
-
 (function (angular) {
+    'use strict';
+
+    var mod = angular.module('viewhead', []);
+
+    // Data service that is shared between all directives
+    mod.factory('viewTitleDataService', function () {
+        return {};
+    });
+
+    mod.directive('viewTitle', ['$rootScope', 'viewTitleDataService', function ($rootScope, viewTitleDataService) {
+        return {
```

Comment:
I was avoiding the local variable on purpose. Ideally the two directives should be splitted into two files and be merged during a build process. In this case you'll have no access to a local variable. But yes, having a publicly accessible service is also not ideal.